### PR TITLE
[ISSUE 3407] sub task4 add the logic of annotation on the splicing class for motan client #3411

### PR DESCRIPTION
**Pull request by ChineseTony**
#3407 add the logic of annotation on the splicing class for motan client;
- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-examples/shenyu-examples-motan/shenyu-examples-motan-service/src/main/java/org/apache/shenyu/examples/motan/service/impl/MotanDemoServiceImpl.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
  * Motan demo service.
  */
 @MotanService(export = "demoMotan:8002")
+@ShenyuMotanClient(path = "/demo/")
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-motan/shenyu-examples-motan-service/src/main/java/org/apache/shenyu/examples/motan/service/impl/MotanDemoServiceImpl.java:28]: Please also consider the situation of `@ShenyuMotanClient(path = "/demo/**")`

Comment (dragon-zhang) [reply shenyu-examples/shenyu-examples-motan/shenyu-examples-motan-service/src/main/java/org/apache/shenyu/examples/motan/service/impl/MotanDemoServiceImpl.java:28]: It is suggested to add a new test class. If you modify the existing test class, you will also need to modify `shenyu-integrated-test-motan` module.

Comment (ChineseTony) [reply shenyu-examples/shenyu-examples-motan/shenyu-examples-motan-service/src/main/java/org/apache/shenyu/examples/motan/service/impl/MotanDemoServiceImpl.java:28]: ok,thanks

File: shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceBeanPostProcessor.java [hunk 2]
```diff
@@ -168,6 +179,39 @@ private MetaDataRegisterDTO buildMetaDataDTO(final Class<?> clazz, final MotanSe
...
+            if (clazz.getInterfaces().length > 0) {
+                serviceName = clazz.getInterfaces()[0].getName();
+            } else {
+                throw new ShenyuClientIllegalArgumentException("Failed to export remote service class " + clazz.getName()
+                        + ", cause: The @Service undefined interfaceClass or interfaceName, and the service class unimplemented any interfaces.");
+            }
+        } else {
+            serviceName = service.interfaceClass().getName();
+        }
+        return MetaDataRegisterDTO.builder()
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceBeanPostProcessor.java:201]: Lost `rpcExt`, which will cause an exception in the call of the `MotanPlugin`. The `metadata` of `motan` and `tars` client seems wrong. I'm discussing it with other developers. Please wait until I have a conclusion before continuing the development.

**Comment by dragon-zhang** [comment]
Please pull the latest code and resolve the conflict.

**Pull request closed without merging**